src/medCore/gui/medAbstractWorkspace.cpp

```diff
@@ -137,6 +137,7 @@ medAbstractWorkspace::~medAbstractWorkspace(void)
 
 void medAbstractWorkspace::addToolBox(medToolBox *toolbox)
 {
+    toolbox->setWorkspace(this);
```

Comment:
To be consistent, removeToolBox should set the workspace to NULL for the toolbox (and so setWorkspace for the toolbox should not make the update connection in case of NULL)